### PR TITLE
Rjurney master docs update

### PR DESCRIPTION
**Pull request by rjurney**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

I've updated the docs for the 0.31 release for developer setup and installation. These directions should work for people on late model Ubuntu/OS X systems. Redo of Pull #7386 against `master` branch.

### ADDITIONAL INFORMATION

- [x] Has associated issue: #7373
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

@mistercrunch
@kristw

### CODE REVIEW

**Comment by rjurney** [comment]
Updated to add a self contained `mysqlclient` back to `requirements-dev.txt`.

**Review by mistercrunch — APPROVED** [review]
This LGTM though some conflicts need to be resolved in order to merge

**Comment by rjurney** [comment]
Conflicts resolved.

File: contrib/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Getting Start with Superset using Docker
```
Comment (mistercrunch) [inline contrib/docker/README.md]: CI is complaining that ASF license is missing from this file

Comment (rjurney) [reply contrib/docker/README.md]: Thanks, fixed.

**Pull request merged**